File: market-domain/src/main/java/com/example/market/domain/inspection/scheduling/InspectionCenterId.java
package com.example.market.domain.inspection.scheduling;

import java.util.Objects;
import java.util.UUID;

public record InspectionCenterId(UUID value) {
    public InspectionCenterId { Objects.requireNonNull(value); }
    public static InspectionCenterId newId() { return new InspectionCenterId(UUID.randomUUID()); }
    public static InspectionCenterId of(String s) { return new InspectionCenterId(UUID.fromString(s)); }
    @Override public String toString() { return value.toString(); }
}

File: market-domain/src/main/kotlin/com/example/market/domain/inspection/scheduling/InspectionCenterId.kt
package com.example.market.domain.inspection.scheduling

import java.util.UUID

data class InspectionCenterId(@get:JvmName("value") val value: UUID) {

    override fun toString(): String = value.toString()

    companion object {
        @JvmStatic
        fun newId(): InspectionCenterId = InspectionCenterId(UUID.randomUUID())

        @JvmStatic
        fun of(s: String): InspectionCenterId = InspectionCenterId(UUID.fromString(s))
    }
}
